codegen-core/src/test/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriterTest.kt

```diff
@@ -99,6 +99,21 @@ class RustWriterTest {
         output shouldContain "/// Top level module"
     }
 
+    @Test
+    fun `normalize HTML`() {
+        val output =
+            normalizeHtml(
+                """
+                <a>Link without href attribute</a>
+                <div>Some text with [brackets]</div>
+                <span>] mismatched [ is escaped too</span>
+                """,
+            )
+        output shouldContain "<code>Link without href attribute</code>"
+        output shouldContain "Some text with \\[brackets\\]"
```

Comment:
Nit: Should we use `"""Some text with \[brackets\]"""` for increased readability?